src/st/no0/57C20.c

```diff
@@ -51,7 +51,538 @@ void func_us_801D91C4(Entity* self) {
+            prim->y0 = prim->y1 = temp_v0_15 - 0x48;
+            prim->priority = self->zPriority;
+            prim->drawMode = DRAW_UNK02;
+
+            prim = prim->next;
+            prim->type = PRIM_TILE;
+            posY = (self->params != 0) << 7;
+            prim->x0 = 0;
+            prim->y0 = posY;
+            prim->v0 = prim->u0 = 0x80;
+            prim->r0 = prim->g0 = prim->b0 = 0;
+
+            prim->priority = 0x11;
+            if (self->params) {
+                prim->priority += 4;
+            }
+
+            prim->drawMode = DRAW_UNK_40 | DRAW_TPAGE | DRAW_TRANSP;
+
+            prim = prim->next;
+            while (prim != NULL) {
+                prim->drawMode = DRAW_HIDE;
+                prim = prim->next;
+            }
+
+            self->zPriority = 0x10;
+
+            if (self->params) {
+                self->zPriority += 4;
+            }
+
+            posY = self->params ? 0x100 : 0x80;
+
+            self->posX.i.hi = 0x20;
+            self->posY.i.hi = posY - 0x28;
+
+            self->palette = *D_us_80180BDE;
+            self->flags &= ~FLAG_POS_CAMERA_LOCKED;
+            self->step_s++;
+            break;
+        case 1:
+            prim = self->ext.prim;
+            self->animCurFrame = 0;
+            prim->type = PRIM_ENV;
+
+            dr_env = LOW(prim->r1);
+            drawEnv = g_CurrentBuffer->draw;
+            drawEnv.isbg = 0;
+            drawEnv.dtd = 0;
+
+            posY = self->params ? 0x180 : 0x100;
+
+            clipRect.x = 0;
+            clipRect.y = posY;
+            clipRect.w = 0x80;
+            clipRect.h = 0x80;
+            drawEnv.clip = clipRect;
+            drawEnv.ofs[0] = 0;
+            drawEnv.ofs[1] = 0x100;
+            SetDrawEnv(dr_env, &drawEnv);
+
+            prim->priority = 0xF;
+            if (self->params) {
+                prim->priority += 4;
+            }
+
+            prim->drawMode = DRAW_DEFAULT;
+            self->ext.et_801D9264.y = 0x28;
+            self->ext.et_801D9264.timer = 0x10;
+            self->step_s++;
+            /* fallthrough */
+        case 2:
+            if (!(g_Timer & 7)) {
+                PlaySfxPositional(0x65B);
+            }
+            prim = self->ext.et_801D9264.unkA4;
+            posX = Random() & 0x3F;
+            posY = self->ext.et_801D9264.y;
+            if (!(g_Timer & 0xF)) {
+                newEntity = AllocEntity(&D_800762D8, (Entity*)&g_EvHwCardEnd);
+                if (newEntity != NULL) {
+                    CreateEntityFromCurrentEntity(2, newEntity);
+                    newEntity->posX.i.hi = prim->x0 + posX;
+                    newEntity->posY.i.hi = prim->y2 + posY - 0x30;
+                    newEntity->params = 3;
+                }
+            }
+            newEntity = AllocEntity(&D_800762D8, (Entity*)&g_EvHwCardEnd);
+            if (newEntity != NULL) {
+                CreateEntityFromCurrentEntity(0x46, newEntity);
```

Comment:
I think by the ID it's a DRA entity? I didn't count but we ought to be able to figure out what this is doing here https://github.com/Xeeynamo/sotn-decomp/blob/6c0544e1c7ee63569cca31b911e8bc5f0f2734fb/src/dra/7A4D0.c#L7

Reply:
`g_DraEntityTbl[]` has size 0x44. `func_us_801D9264` uses entity ids 0x44 and 0x45. Could it be referring to entities elsewhere ?

Reply:
Hmm not sure what's going on then, the entity lookup code is here https://github.com/Xeeynamo/sotn-decomp/blob/6c0544e1c7ee63569cca31b911e8bc5f0f2734fb/src/dra/7A4D0.c#L97

Reply:
Could `g_DraEntityTbl` be incomplete ? This function uses `g_DraEntityTbl` if entity id is between 0 and 0xCF.

Reply:
The data after the table is blueprints https://github.com/Xeeynamo/sotn-decomp/blob/b978d7e432e4fe17b07704df1f7f6f8b8626cf2c/config/splat.us.dra.yaml#L107
Maybe it's wrong? It could also just be an out of bounds bug or something I'm not noticing

Reply:
Another thing is that g_DraEntityTbl may have `NULL`s at the end that are being padded as zeroes by the compiler at the moment